### PR TITLE
Clean up the grub2 reinstall entries in autoyast minion snippet

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

See title

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- No tests: reinstallation autoyast is not easy to automatically test

- [X] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by cbosdo** [comment]
@aaannz could you take a look at this PR?

**Review by aaannz — APPROVED** [review]
LGTM, but is this enough? Looking at [autoinstall.sls](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/bootloader/autoinstall.sls) we support grub, elilo and grub2.

**Comment by SchoolGuy** [comment]
@cbosdo Sorry that I have not yet seen this. I don't know how it slipped my notifications. What I am wondering is how the entry for reinstallation goes there in the first place. Wouldn't it be good to prevent it from being created? And if this is intentional (as it seems in the linked sls file), then can't we use Salt to remove the entry again?

@aaannz Is GRUB for us still relevant? I thought that we only have GRUB2 nowadays?

To sum up my opinion about this PR: I am not against it, just a bit of questionmarks on the reason for the change.

**Comment by aaannz** [comment]
> @aaannz Is GRUB for us still relevant? I thought that we only have GRUB2 nowadays?
> 
I did some archeology and support for grub and elilo was added by @vzhestkov as a response to [bsc#1164227](https://bugzilla.suse.com/show_bug.cgi?id=1164227) which is all about SLE11 series. So you are right, we do not need it anymore and IMO we should remove it from autoinstallation.sls and make it grub2 only again.

File: java/spacewalk-java.changes [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Clean grub2 reinstall entry in autoyast snippet (bsc#119950)
```
Comment (aaannz) [inline java/spacewalk-java.changes:1]: This is wrong bug number - https://bugzilla.suse.com/show_bug.cgi?id=1199950
```suggestion
- Clean grub2 reinstall entry in autoyast snippet (bsc#1199950)
```

**Review by aaannz — CHANGES_REQUESTED** [review]

File: java/spacewalk-java.changes (hunk 1 above)
Comment (cbosdo) [reply java/spacewalk-java.changes]: addressed

**Comment by cbosdo** [comment]
> And if this is intentional (as it seems in the linked sls file), then can't we use Salt to remove the entry again?

That would probably mean changing the whole script in that autoyast snippet to use Salt since it runs at the very end of the auto-installation... what is the value added of a refactoring to use Salt here? I wouldn't do it for such a bug fix PR

**Review by SchoolGuy — APPROVED** [review]
LGTM. Indeed for a bugfix there is no value refactoring this atm. However if we use salt, we should use it in the long run.
